sources/Algo/src/DynModelFilterAlgorithm.cpp

```diff
@@ -0,0 +1,126 @@
+#include "Constants.h"
+#include "Log.h"
+
+namespace dfl {
+namespace algo {
+
+void
+DynModelFilterAlgorithm::filter() {
+  removeRpclInGeneratorsAndSvcIfMissingConnexionToSvc();
+  filterPartiallyConnectedDynamicModels();
+}
+
+void
+DynModelFilterAlgorithm::removeRpclInGeneratorsAndSvcIfMissingConnexionToSvc() {
+  std::vector<DynamicModelDefinition::DynModelId> svcToRemove;
+  for (const auto& dynamicModel : dynamicModelsToFilter_) {
+    if (dynamicModel.second.lib == dfl::common::constants::svcModelName) {
+      bool isSVCconnectedToUMeasurement = checkIfSVCconnectedToUMeasurement(dynamicModel.second);
+      if (!isSVCconnectedToUMeasurement) {
+        svcToRemove.push_back(dynamicModel.second.id);
+        for (const DynamicModelDefinition::MacroConnection& generatorConnectedToSVC : dynamicModel.second.nodeConnections) {
+          const DynamicModelDefinition::MacroConnection::ElementId& genId = generatorConnectedToSVC.connectedElementId;
+          auto foundGen = std::find_if(generators_.begin(), generators_.end(),
+                                        [&genId](const algo::GeneratorDefinition& genDefinition) {
+                                          return genDefinition.id == genId;
+                                        });
+          if (foundGen != generators_.end()) {
+            foundGen->removeRpclFromModel();
+          }
+        }
+      }
+    }
+  }
+  for (const DynamicModelDefinition::DynModelId& svcId : svcToRemove) {
+    dynamicModelsToFilter_.erase(svcId);
+  }
+}
+
+bool
+DynModelFilterAlgorithm::checkIfSVCconnectedToUMeasurement(const algo::DynamicModelDefinition& svcModel) const {
+  bool isSVCconnectedToUMeasurement = false;
+  const std::map<std::string, inputs::AssemblingDataBase::DynamicAutomaton>& assemblingAutomatons = assembling_.dynamicAutomatons();
+  auto assemblingAutomatonIt = assemblingAutomatons.find(svcModel.id);
+  if (assemblingAutomatonIt != assemblingAutomatons.end()) {
+    const std::vector<inputs::AssemblingDataBase::MacroConnect>& macroConnects = assemblingAutomatonIt->second.macroConnects;
+    for (inputs::AssemblingDataBase::MacroConnect macroConnect : macroConnects) {
+      const inputs::AssemblingDataBase::SingleAssociation& singleAssociation = assembling_.getSingleAssociation(macroConnect.id);
+      if (singleAssociation.bus.is_initialized()) {
+        auto macroConnectIt = std::find_if(svcModel.nodeConnections.begin(), svcModel.nodeConnections.end(),
+                                            [&macroConnect](const DynamicModelDefinition::MacroConnection& nodeConnection) {
+                                              return nodeConnection.id == macroConnect.macroConnection;
+                                            });
+        if (macroConnectIt != svcModel.nodeConnections.end()) {
+          isSVCconnectedToUMeasurement = true;
+          break;
+        }
+      }
+    }
+  } else {
+    throw Error(UnknownAutomaton, svcModel.id);
```

Comment:
Should never happen

Reply:
It may be not safe to remove this exception

Reply:
an assert should be enough: if this happens it means that we have "invented" an automaton that was not in the assembling